components/net/cookie_storage.rs

```diff
@@ -37,6 +37,23 @@ impl CookieStorage {
         let domain = reg_host(cookie.cookie.domain.as_ref().unwrap_or(&"".to_string()));
         let cookies = self.cookies_map.entry(domain).or_insert(vec![]);
 
+        // https://www.ietf.org/id/draft-ietf-httpbis-cookie-alone-01.txt Step 2
+        if !cookie.cookie.secure && !source.is_secure_protocol() &&
+           cookies.iter().any(|c| {
+                let path_match = if let (&Some(ref new), &Some(ref existing)) = (&cookie.cookie.path, &c.cookie.path) {
+                    Cookie::path_match(new, existing)
+                } else {
+                    cookie.cookie.path == c.cookie.path
+                };
```

Comment:
Our cookies always have a path according to https://github.com/KiChjang/servo/blob/cf08600e5f390b72a6e094e74633326bad204908/components/net/cookie.rs#L69-L74, so let's use expect on both of them instead.